Whisper fails if language of event is set

**Describe the bug**
Once the language attribute of an event has been set, Whisper will no longer be able to generate subtitles for that event. Opencast is using the wrong language format for the Whisper command. In the logs I can see that the `--language eng` option is used, for the command to work the 2-digit language code must be used: `--language en`.

For an event where the language hasn't been set, Whisper will automatically detect the language.

**To Reproduce**

Steps to reproduce the behavior:
1. Create an event where the language attribute is set
2. Start a workflow to generate captions with whisper
3. See error


**Expected behavior**

Subtitles can also be generated with Whisper if the event language attribute is set.

**Stack trace**

```
2024-06-20T16:13:32,532 | DEBUG | (WhisperEngine:231) - Transcription failed closing Whisper transcription process for: /srv/opencast/workspace/https_admin.opencast-test/assets/assets/2ac051ea-49d0-4af2-90d5-3b07d904e1fa/0579b122-42e0-41d5-b9e7-21a18454b669/4/Test-2min.mp4
2024-06-20T16:13:32,589 | ERROR | (AbstractJobProducer$JobRunner:343) - Error handling operation 'speechtotext':
org.opencastproject.speechtotext.api.SpeechToTextServiceException: Error while generating subtitle from https://admin.opencast-test/assets/assets/2ac051ea-49d0-4af2-90d5-3b07d904e1fa/0579b122-42e0-41d5-b9e7-21a18454b669/4/Test-2min.mp4
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:165) ~[?:?]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:313) [!/:?]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:272) [!/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.opencastproject.speechtotext.api.SpeechToTextEngineException: org.opencastproject.speechtotext.api.SpeechToTextEngineException: Whisper exited abnormally with status 2 (command: [/opt/whisper/venv/bin/whisper-ctranslate2, /srv/opencast/workspace/https_admin.opencast-test/assets/assets/2ac051ea-49d0-4af2-90d5-3b07d904e1fa/0579b122-42e0-41d5-b9e7-21a18454b669/4/Test-2min.mp4, --model, medium, --output_dir, /srv/opencast/workspace/collection/subtitles/job-801201, --language, eng, --model_dir, /var/lib/opencast/whisper-models, --vad_filter, True, --device, cpu, --compute_type, int8])
	at org.opencastproject.speechtotext.impl.engine.WhisperEngine.generateSubtitlesFile(WhisperEngine.java:232) ~[?:?]
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:154) ~[?:?]
	... 6 more
Caused by: org.opencastproject.speechtotext.api.SpeechToTextEngineException: Whisper exited abnormally with status 2 (command: [/opt/whisper/venv/bin/whisper-ctranslate2, /srv/opencast/workspace/https_admin.opencast-test/assets/assets/2ac051ea-49d0-4af2-90d5-3b07d904e1fa/0579b122-42e0-41d5-b9e7-21a18454b669/4/Test-2min.mp4, --model, medium, --output_dir, /srv/opencast/workspace/collection/subtitles/job-801201, --language, eng, --model_dir, /var/lib/opencast/whisper-models, --vad_filter, True, --device, cpu, --compute_type, int8])
	at org.opencastproject.speechtotext.impl.engine.WhisperEngine.generateSubtitlesFile(WhisperEngine.java:218) ~[?:?]
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:154) ~[?:?]
	... 6 more
```

**Server environment:**
 - Version 15

--- a/modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperEngine.java
+++ b/modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperEngine.java
@@ -106,6 +106,9 @@ public class WhisperEngine implements SpeechToTextEngine {
   /** Map to get ISO 639 language code for language name in English */
   private Map<String, String> languageMap = new HashMap<>();
 
+  /** Map to get ISO 639 language code for 3-letter language code */
+  private Map<String, String> languageISO3to2Map = new HashMap<>();
+
   @Override
   public String getEngineName() {
     return engineName;
@@ -136,6 +139,8 @@ public class WhisperEngine implements SpeechToTextEngine {
       Locale locale = new Locale(languageCode);
       String languageName = locale.getDisplayLanguage(new Locale("en"));
       languageMap.put(languageName, languageCode);
+      String languageISO3 = locale.getISO3Language();
+      languageISO3to2Map.put(languageISO3, languageCode);
     }
     logger.debug("Filled language map.");
 
@@ -170,9 +175,15 @@ public class WhisperEngine implements SpeechToTextEngine {
     }
 
     if (!language.isBlank() && !translate) {
-      logger.debug("Using language {} from workflows", language);
-      transcriptionCommand.add("--language");
-      transcriptionCommand.add(language);
+      // get 2-letter language code
+      language = languageISO3to2Map.getOrDefault(language,"");
+      if (language.isBlank()) {
+        logger.warn("No 2-letter language code found, using language auto detection");
+      } else {
+        logger.debug("Using language {} from workflows", language);
+        transcriptionCommand.add("--language");
+        transcriptionCommand.add(language);
+      }
     }
 
     if (quantization != null) {
@@ -259,4 +270,3 @@ public class WhisperEngine implements SpeechToTextEngine {
     return new Result(language, output);
   }
 }
-

